cogs/slowmode.py

```diff
@@ -21,14 +41,57 @@ def __init__(self, client):
     @commands.has_any_role(MOD_ROLE, HELPER_ROLE)
     async def slowmode(self, ctx: commands.Context, interval_str: str):
         '''Activate slowmode in help channels'''
-        # TODO: check against channel ID instead of name
         interval = util.get_id_from_tag(interval_str)
-        if isinstance(ctx.channel, discord.TextChannel) and \
-                ctx.channel.name in ['kb-help', 'kb-buying-advice']:
+        if isinstance(ctx.channel, discord.TextChannel) and ctx.channel.id in CHANNELS:
             await ctx.channel.edit(slowmode_delay=interval)
             await ctx.channel.send("Slow it down!")
 
 
+    def get_delay(self, message_count):
+        message_limits = sorted(time_configs, reverse=True)
+        for limit in message_limits:
+            if message_count >= limit:
+                return time_configs[limit]
+        return 0
+
+
+    async def update_slowmode(self):
+        global last_updated, message_cache, previous_delays
+        new_channel_delays = {}
+        for channel_id in message_cache.keys():
+            delay = self.get_delay(message_cache[channel_id])
+            if channel_id in previous_delays.keys():
+                if previous_delays[channel_id] == delay:
+                    new_channel_delays[channel_id] = delay
+                    continue
+            channel = self.client.get_channel(channel_id)
+            await channel.edit(slowmode_delay=delay)
+            await channel.send(f"Slowmode has been automaticly set to {delay}s")
+            new_channel_delays[channel_id] = delay
+
+        message_cache = {}
+        last_updated = time.time()
+        previous_delays = new_channel_delays
+
+
+    @commands.event
+    async def on_message(self, message):
+        global last_updated, message_cache
+        channel_id = message.channel.id
+
+        if time.time() >= last_updated + UPDATE_FREQUENCY:
+            await self.update_slowmode()
+
+        if channel_id not in CHANNELS:
+            return
```

Comment:
i feel like this should be at the top of the function, return as early as possible if from irrelevant channel

Reply:
moved if statement to top of function